src/User.php

```diff
@@ -6581,6 +6581,13 @@ public static function getIcon()
     public function applyGroupsRules()
     {
         if (!isset($this->input["_ldap_rules"]['groups_id'])) {
+            if (isset($this->input["_ldap_rules"]) && isset($this->input['id'])) {
+                $group_user = new Group_User();
+                $groups = $group_user->find(['users_id' => $this->input['id'], 'is_dynamic' => 1]);
+                foreach ($groups as $group) {
+                    $group_user->delete($group);
```

Comment:
If the rule action is modified to assign another group, the old dynamic group is not deleted?

Reply:
Yes, but can you add this case to the tests?

In addition, I think that the group checks are not precise enough, because we count without checking that they are the right groups.

Reply:
You can perhaps use `isUserInGroup()` to check